Commands/Public/safebooru.js

```diff
@@ -0,0 +1,74 @@
+const ArgParser = require("../../Modules/MessageUtils/Parser");
+const PaginatedEmbed = require("../../Modules/MessageUtils/PaginatedEmbed");
+const fetch = require("chainfetch");
+
+module.exports = async ({ Constants: { Colors, Text, APIs, UserAgent, NSFWEmbed }, client }, { serverDocument }, msg, commandData) => {
+	if (!msg.channel.nsfw) {
+		return msg.send(NSFWEmbed);
+	}
+	if (!msg.suffix) {
+		return msg.sendInvalidUsage(commandData, "You gotta give me somethin' to search for!");
+	}
+
+	let num;
+	const args = ArgParser.parseQuoteArgs(msg.suffix, msg.suffix.includes("|") ? "|" : " ");
```

Comment:
*cracks knuckles* Ok. I'm all fine, but I had to bring this up.

There are certain tags we MUST filter if we don't want ToS issues, even for NSFW channels/images. Can we implement a check if you provide any of those tags, we throw an error, and if any post contains those tags we don't include it?

A short list might be: `['loli', 'shota', 'cub', 'young', 'child', 'baby', 'guro', 'gore', 'vore', 'cheese_grater']`. We can add more as needed, probably make them a JSON file

Reply:
I'm not sure we can filter them in the API request, but we could potentially filter the results on our side. This however means that when such a post is filtered, the end result count is lower than what the user requested/the content fetching default.

Regardless, a very good catch. This could really give us issues if not handled. I'm thinking we use Constants.js here. We should probably create lists like these for all NSFW commands.

Reply:
This list applies to all NSFW commands, not just one or the other. If we filter `shit` in one command we should filter it across all commands

Reply:
True. Coincidentally, all our 3 NSFW APIs are pretty similar when it comes to tags, so this will be easy to implement globally. (Although, in theory, Safebooru shouldn't be porn-NSFW) Also, it might be worth implementing a similar filter for queries in other content fetching commands? (Google, image, imgur, gif, etc.)